spec/services/search_manifestations_spec.rb

```diff
@@ -438,127 +676,269 @@
 
       context "when 'from' and 'to' values are equal" do
         let(:range) { { 'from' => 1950, 'to' => 1950 } }
+
+        before do
+          Chewy.strategy(:atomic) do
+            create(:manifestation, work_date: '15.06.1950')
+            create(:manifestation, work_date: '31.12.1950')
+            create(:manifestation, work_date: '31.12.1949')
+            create(:manifestation, work_date: '01.01.1951')
+          end
+        end
+
         it 'returns all records created in given year' do
-          assert_date_range(4)
+          assert_date_range(2)
         end
       end
 
       context "when 'from' and 'to' values are different" do
         let(:range) { { 'from' => 1950, 'to' => 1952 } }
+
+        before do
+          Chewy.strategy(:atomic) do
+            create(:manifestation, work_date: '01.01.1950')
+            create(:manifestation, work_date: '15.06.1951')
+            create(:manifestation, work_date: '31.12.1952')
+            create(:manifestation, work_date: '31.12.1949')
+            create(:manifestation, work_date: '01.01.1953')
+          end
+        end
+
         it "returns all records created from beginning of 'from' to end of 'to' year" do
-          assert_date_range(12)
+          assert_date_range(3)
         end
       end
 
       context "when only 'from' value provided" do
         let(:range) { { 'from' => 1985 } }
+
+        before do
+          Chewy.strategy(:atomic) do
+            create(:manifestation, work_date: '01.01.1985')
+            create(:manifestation, work_date: '15.06.1990')
+            create(:manifestation, work_date: '31.12.1984')
+          end
+        end
+
         it 'returns all records created starting from given year' do
-          assert_date_range(60)
+          assert_date_range(2)
         end
       end
 
       context "when only 'to' value provided" do
         let(:range) { { 'to' => 1952 } }
+
+        before do
+          Chewy.strategy(:atomic) do
+            create(:manifestation, work_date: '15.06.1952')
+            create(:manifestation, work_date: '31.12.1951')
+            create(:manifestation, work_date: '01.01.1953')
+          end
+        end
+
         it 'returns all records created before or in given year' do
-          assert_date_range(12)
+          assert_date_range(2)
         end
       end
     end
   end
 
   describe 'sorting' do
-    let!(:subject) do
-      SearchManifestations.call(sorting, sort_dir, {}).limit(REC_COUNT).map(&:id)
-    end
-    let(:asc_order) { Manifestation.all_published.order(db_column => :asc).pluck(:id) }
-    let(:desc_order) { asc_order.reverse }
-
     describe 'alphabetical' do
-      let(:db_column) { :sort_title }
       let(:sorting) { 'alphabetical' }
+      let!(:manifestation_a) { nil }
+      let!(:manifestation_b) { nil }
+      let!(:manifestation_c) { nil }
+
+      before do
+        Chewy.strategy(:atomic) do
+          @manifestation_a = create(:manifestation, title: 'Apple')
```

Comment:
don't use member variables for test data. Use let block instead.
Update all such places.